integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java

```diff
@@ -8,26 +8,42 @@
 import io.openlineage.spark.agent.facets.SparkPropertyFacet;
 import io.openlineage.spark.api.CustomFacetBuilder;
 import io.openlineage.spark.api.OpenLineageContext;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 import java.util.function.BiConsumer;
 import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.spark.SparkConf;
+import org.apache.spark.SparkContext;
+import org.apache.spark.scheduler.SparkListenerEvent;
 import org.apache.spark.scheduler.SparkListenerJobStart;
+import org.apache.spark.sql.SparkSession;
 
+@Slf4j
 public class SparkPropertyFacetBuilder
-    extends CustomFacetBuilder<SparkListenerJobStart, SparkPropertyFacet> {
+    extends CustomFacetBuilder<SparkListenerEvent, SparkPropertyFacet> {
   private static final Set<String> DEFAULT_ALLOWED_PROPERTIES =
       new HashSet<>(Arrays.asList("spark.master", "spark.app.name"));
   private static final String ALLOWED_PROPERTIES_KEY = "spark.openlineage.capturedProperties";
-  private final SparkConf conf;
-  private final Set<String> allowerProperties;
+  private SparkConf conf;
+  private Set<String> allowerProperties;
 
   public SparkPropertyFacetBuilder(OpenLineageContext context) {
-    conf = context.getSparkContext().getConf();
+    fillConfAndAllowerProperties(context.getSparkContext());
+  }
+
+  public SparkPropertyFacetBuilder() {
+    try {
+      SparkSession session = SparkSession.active();
```

Comment:
Please consider moving whole logic to `build` method. It would make logic more consistent when having a single `SparkSession.active()` call. Wouldn't it?

Reply:
@pawel-big-lebowski Sorry again for replying your messages too late.
Yes, you're right. We have called SparkSession.active() two times in this file. That is because the SparkPropertyFacetBuilder class has two properties (conf and allowerProperties). We have to initialize them in the constructor method. So, the first time we called spark session.active() is to get the value of 'spark.openlineage.capturedProperties', while the second time we called this is to get the run-time value of the keys in allowerProperties. I think they are different and hard to be merged together.